python/lsst/pipe/tasks/insertFakeSources.py

```diff
@@ -0,0 +1,597 @@
+
+            try:
+                psfKernel = psf.computeKernelImage(xy).getArray()
+            except InvalidParameterError:
+                self.log.info("Galaxy at %0.4f, %0.4f outside of image" % (row["x"], row["y"]))
+                continue
+            psfIm = galsim.InterpolatedImage(galsim.Image(psfKernel), scale=pixelScale)
+            gal = galsim.Convolve([gal, psfIm])
+            try:
+                galIm = gal.drawImage(scale=pixelScale, method="real_space").array
+            except (galsim.errors.GalSimFFTSizeError, MemoryError):
+                continue
+
+            galImages.append((afwImage.ImageF(galIm), xy))
+
+        return galImages
+
+    def mkFakeStars(self, fakeCat, band, photoCalib, psf, calexp):
+
+        """Make fake stars based off the properties in the fakeCat.
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+        band : `str`
+        photoCalib : `lsst.afw.image.calib.Calib` or `lsst.afw.image.photoCalib.PhotoCalib`
+        psf : `lsst.meas.extensions.psfex.psfexPsf.PsfexPsf`
+
+        Returns
+        -------
+        starImages : `list`
+            A list of `lsst.afw.image.image.image.ImageF` of fake stars
+        """
+
+        starImages = []
+
+        self.log.info("Making %d fake star images" % len(fakeCat))
+
+        for (index, row) in fakeCat.iterrows():
+            # jointCal photoCalib and normal photoCalib have different ways of doing this.
+            xy = afwGeom.Point2D(row["x"], row["y"])
+            if self.config.useUpdatedCalibs and self.calexpType == "calexp":
+                # This should go away when this is switched to reference catalogs
+                try:
+                    flux = photoCalib.magnitudeToInstFlux(row[band + "magVar"], xy)
+                except LogicError:
+                    flux = 0
+            else:
+                flux = photoCalib.getFlux(row[band + "magVar"])
+
+            try:
+                starIm = psf.computeImage(xy)
+
+            except InvalidParameterError:
+                self.log.info("Star at %0.4f, %0.4f outside of image" % (row["x"], row["y"]))
+                continue
+
+            starIm *= flux
+            starImages.append((starIm.convertF(), xy))
+
+        return starImages
+
+    def cleanCat(self, fakeCat):
+        """Remove rows from the fakes catalog which have HLR = 0 for either the buldge or disk component
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+
+        Returns
+        -------
+        fakeCat : `pandas.core.frame.DataFrame`
+        """
+
+        goodRows = ((fakeCat[self.config.bulgeHLR] != 0.0) & (fakeCat[self.config.diskHLR] != 0.0))
+
+        badRows = len(fakeCat) - len(goodRows)
+        self.log.info("Removing %d rows with HLR = 0 for either the bulge or disk" % badRows)
+
+        return fakeCat[goodRows]
+
+    def addFakeSources(self, calexp, fakeImages, sourceType):
```

Comment:
If mkGalsimGalaxies and mkFakeStars took only parameters and did not loop internally and make lists, you could call them in one (or two for each type) here and avoid the memory overhead of storing  all of them at once, but I really dont know how much memory that ends up taking up to judge if it is worth refactoring the code. Something for you to consider, but up to you what you think is best

Reply:
Not much